src/test/scala/cognite/spark/v1/AlphaDataModelInstancesRelationTest.scala

```diff
@@ -709,51 +779,180 @@ class DataModelInstancesRelationTest
+        val df = readRows(primEdgeExtId, metricPrefix)
+        df.limit(1).count() shouldBe 1
+        getNumberOfRowsRead(metricPrefix, "alphadatamodelinstances") shouldBe 1
+        val data = df.collect()
+        data.head.getAs[Float]("prop_float") shouldBe 2.1
+        data.head.getAs[String]("type") shouldBe "test2"
+        data.head.getAs[Boolean]("prop_bool") shouldBe false
+        data.head.getAs[String]("prop_string") shouldBe "abc"
+      }
+    )
+  }
+
+  it should "read edges with special property types" in {
+    val randomId = "prim_edge_test_" + shortRandomString()
+    val randomId2 = "prim_edge_test_" + shortRandomString() + "_2"
+    tryTestAndCleanUp(
+      Seq(randomId, randomId2), {
+        retryWhile[Boolean](
+          {
+            Try {
+              insertRows(
+                specialEdge,
+                spark
+                  .sql(
+                    s"""
+                       |select 'testNode1' as startNode,
+                       |'testNode3' as endNode,
+                       |'test1' as type,
+                       |'asset' as prop_direct_relation,
+                       |timestamp('2022-01-01 13:34:56.789') as prop_timestamp,
+                       |date('2022-01-20') as prop_date,
+                       |'${randomId}' as externalId
+                       |
+                       |union all
+                       |
+                       |select 'testNode1' as startNode,
+                       |'testNode2' as endNode,
+                       |'test2' as type,
+                       |'asset2' as prop_direct_relation,
+                       |timestamp('2022-01-10 13:34:56.789') as prop_timestamp,
+                       |date('2022-01-01') as prop_date,
+                       |'${randomId2}' as externalId""".stripMargin)
+              )
+            }.isFailure
+          },
+          failure => failure
+        )
+        val metricPrefix = shortRandomString()
+        val df = readRows(specialEdge, metricPrefix)
+        df.where("endNode = 'testNode3'").count() shouldBe 1
+        getNumberOfRowsRead(metricPrefix, "alphadatamodelinstances") shouldBe 1
+
+        val metricPrefix2 = shortRandomString()
+        val data = readRows(specialEdge, metricPrefix2)
+          .where("timestamp('2022-01-01 13:34:56.789') = prop_timestamp and prop_date > date('2022-01-02')")
+          .collect()
+        getNumberOfRowsRead(metricPrefix2, "alphadatamodelinstances") shouldBe 1
+        data.length shouldBe 1
+        data.head.getAs[String]("prop_direct_relation") shouldBe "asset"
+        data.head.getAs[String]("startNode") shouldBe "testNode1"
+        data.head.getAs[String]("endNode") shouldBe "testNode3"
+        data.head.getAs[String]("type") shouldBe "test1"
+        data.head.getAs[java.sql.Timestamp]("prop_timestamp") shouldBe java.sql.Timestamp.valueOf("2022-01-01 13:34:56.789")
+        data.head.getAs[java.sql.Date]("prop_date") shouldBe java.sql.Date.valueOf("2022-01-20")
```

Comment:
```scala
data.headOption.map(_.getAs[type])(propName) shouldBe Some(...)
```

Reply:
thanks  I have just seen you left a review, I'll fix things.